math/mode.go

```diff
@@ -0,0 +1,48 @@
+// mode.go
+// author(s): [CalvinNJK] (https://github.com/CalvinNJK)
+// description: Finding Mode Value In an Array
+// see mode.go
+
+package math
+
+import (
+	"errors"
+
+	"github.com/TheAlgorithms/Go/constraints"
+)
+
+// Error when array is empty
```

Comment:
Just one last thing, everything else seems ok to me now.

```suggestion
// ErrEmptySlice is the error returned by functions in math package when
// an empty slice is provided to it as argument when the function expects
// a non-empty slice.
```